### PR TITLE
just a few checks to be extra sure

### PR DESCRIPTION
**Pull request by kevinkreiser**
chef  uses the elevation_extract script to load data onto an aws machine. these checks make sure the data is good.

### CODE REVIEW

File: scripts/elevation_extract.sh [hunk 1]
```diff
@@ -44,26 +44,22 @@ for x in $(seq ${min_x} 1 ${max_x}); do
 		file=$(python -c "print '%s%02d%s%03d.hgt' % ('S' if ${y} < 0 else 'N', abs(${y}), 'W' if ${x} < 0 else 'E', abs(${x}))")
                 dir=$(echo ${file} | sed "s/^\([NS][0-9]\{2\}\).*/\1/g")
 		#if we dont already have it or its the wrong size
-		if [ ! -e ${dest}/${dir}/${file} ] || [ $(wc -c < ${dest}/${dir}/${file}) -ne $((3601*3601*2)) ]; then
+		if [ ! -e ${dest}/${dir}/${file} ] || [ $(stat -c %s ${dest}/${dir}/${file}) -ne $((3601*3601*2)) ]; then
```
Comment (kevinkreiser) [inline scripts/elevation_extract.sh]: this allows picking up where you left off if some tiles didnt properly download last time

File: scripts/elevation_extract.sh [hunk 2]
```diff
@@ -44,26 +44,22 @@ for x in $(seq ${min_x} 1 ${max_x}); do
...
-			exit 1
-		fi
-		#wrong size
-		if [ $(wc -c < ${dest}/${dir}/${file}) -ne $((3601*3601*2)) ]; then
-			echo "${dest}/${dir}/${file} is corrupt"
-			exit 1
-		fi
-	done
-done
+total=$(find ${dest} | grep -cF hgt)
```
Comment (kevinkreiser) [inline scripts/elevation_extract.sh]: from here down we make sure we have the right number of files and that they are all the right size

File: src/skadi/sample.cc [hunk 3]
```diff
@@ -69,8 +76,12 @@ namespace skadi {
     auto files = get_files(data_source);
     for(const auto& f : files) {
       auto index = is_hgt(f);
-      if(index < cache.size())
-        cache[index].map(f, HGT_PIXELS * sizeof(int16_t));
+      if(index < cache.size()) {
+        if(file_size(f) == HGT_PIXELS * sizeof(int16_t))
```
Comment (kevinkreiser) [inline src/skadi/sample.cc]: previously we blindly memmaped a file a particular size assuming it was good. now we dont map it and dont use it if its not the right size, and we warn about it

**Comment by dgearhart** [comment]
:shipit:

**Pull request merged**